### DISCUSSION TITLE
Allow responsive sizing at breakpoints for Cards

### DISCUSSION THREAD

**Discussion by RhysLees** [laravel/nova-issues#4155] [Ideas & Feature Requests]
It would be awesome now that Nova 4 is responsive to be able to change the width of card breakpoints.

Now sure what the best way would be to implement it but maybe something along the lines of allowing  `->width()` to be passed an array with tailwind keys for the breakpoint e.g:

```php
(new EnquiriesPerDay)->width([
    'sm' => Card::FULL_WIDTH,
    'md' => Card::TWO_THIRDS_WIDTH,
    'lg' => Card::ONE_HALF_WIDTH,
    'xl' => Card::ONE_QUARTER_WIDTH
]),
```

or having separate functions for the breakpoints:

```php
(new EnquiriesPerDay)
    ->widthSmall(Card::FULL_WIDTH),
    ->widthMedium(Card::TWO_THIRDS_WIDTH),
    ->widthLarge(Card::ONE_HALF_WIDTH),
    ->widthExtraLarge( Card::ONE_QUARTER_WIDTH),
```

Currently, I don't believe that the following sizing can be utilised as much as they could as at some breakpoints the smaller sizing breaks UI and as some of the metric cards specific overlap with other elements on the card forcing us to use larger sizing

```php
    public const ONE_THIRD_WIDTH = '1/3';
    public const ONE_QUARTER_WIDTH = '1/4';
```

![image](https://user-images.githubusercontent.com/43909932/166562309-4392e62a-7ca1-48d7-bf8a-15639e0f16fb.png)

https://user-images.githubusercontent.com/43909932/166562532-c4f49bab-4d31-4348-a115-399c1fa86a8c.mp4

Its ok once it goes fully mobile layout and forced full width but inbeween has issues of overlaping.

**Comment by xfolder** [comment]
Up

1920px
![image](https://user-images.githubusercontent.com/13942856/191232498-91e83c50-c768-42bd-97c7-f910053feb78.png)

1024
![image](https://user-images.githubusercontent.com/13942856/191232593-8a3d54b7-922f-48b1-b884-17623e9f893e.png)

**Comment by corgalore** [comment]
Please, responsive cards is necessary to keep a clean structure without resorting to custom css to override the nova classes.
